### PR TITLE
chore: Test nullable lists in Terraform

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
## Changes
- Add two resources testing the Terraform SDKv2 ability to differentiate between filled, empty, and null lists:
  - Just regular resource with list
  - Resource with additional computed field which aids in empty <-> null transitions
- Add a set of acceptance tests that check for state transitions in both resources, and checks how the external changes are handled

### CODE REVIEW

File: pkg/testfunctional/test_resource_list_null_value_logic_acceptance_test.go [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+			Steps: []resource.TestStep{
+				{
+					PreConfig: func() { t.Setenv(envName, "") },
+					Config:    configWithEmpty,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr(ref, "nullable_list.#", "0"),
+					),
+				},
+				{
+					PreConfig: func() { t.Setenv(envName, "__NULL__") },
```
Comment (sfc-gh-asawicki) [inline pkg/testfunctional/test_resource_list_null_value_logic_acceptance_test.go:436]: nit: can we use extracted values for "__NULL__" etc?

File: pkg/acceptance/testenvs/testing_environment_variables.go [hunk 2]
```diff
@@ -45,6 +45,7 @@ const (
 
 	SimplifiedIntegrationTestsSetup env = "TEST_SF_TF_SIMPLIFIED_INTEGRATION_TESTS_SETUP"
 
+	TestResourceNullListHandlingEnv     env = "TEST_SF_TF_TEST_RESOURCE_DATA_NULL_LIST_HANDLING_ENV"
```
Comment (sfc-gh-asawicki) [inline pkg/acceptance/testenvs/testing_environment_variables.go:48]: nit: why env and not existing http server helper?

Comment (sfc-gh-jcieslak) [reply pkg/acceptance/testenvs/testing_environment_variables.go:48]: tbh, forgot that we could use it here, I was looking at other cases where env-way was still used

File: pkg/testfunctional/test_resource_list_null_value_logic_acceptance_test.go (hunk 1 above)
Comment (sfc-gh-jcieslak) [reply pkg/testfunctional/test_resource_list_null_value_logic_acceptance_test.go:436]: You mean constants I defined in the resource file?

**Comment by sfc-gh-jcieslak** [comment]
Merging, but can address nits follow-up (if needed)
